### ISSUE TITLE
 Enable TOF support for blocks geometry in SAFIR_genericScanner

### ISSUE THREAD

**Issue by chengqi001** [UCL/STIR#1506]
**This issue requests the addition of TOF (Time-of-Flight) support for the blocks geometry in the SAFIR_genericScanner example.**
1. Currently, TOF functionality is available only for cylindrical geometries.
2. The blocks geometry in the SAFIR listmode does not yet support TOF.
3. The SAFIR listmode format currently does not contain TOF information, and adding this information is necessary.

**Comment by chengqi001** [comment]
A Feasible Step:
1. Adapt TOF for the Blocks Scanner
2. Transfer TOF implementation to SAFIR
3. Convert SAFIR's TOF (recommended to be stored in ‘ unsigned reserved : 6 ’) to sinograms

**Comment by KrisThielemans** [comment]
For the first step, the relevant issue is https://github.com/UCL/STIR/issues/1307. This isn't a trivial thing sadly.

I assume that adapting the SAFIR reader itself isn't too hard, but didn't really check. I suppose this could be done independently from the "blocks" adaptation (as SAFIR would read "cylindrical" scanner data as well). Currently, I'm afraid this isn't on our task list yet, but contributions would be very welcome (and checked). Feel free to ask for help here.

**Comment by KrisThielemans** [comment]
Another difficult turns out to be #1077

**Comment by chengqi001** [comment]
> Another difficult turns out to be [#1077](https://github.com/UCL/STIR/issues/1077)
Alright, I have responded to issue #1077. Thank you!

**Comment by KrisThielemans** [comment]
Note that there is a related email discussion (TOF for SAFIR for cylindrical scanner) at https://sourceforge.net/p/stir/mailman/stir-users/thread/BYAPR02MB42326FE5753470D1458A89ABC83F2%40BYAPR02MB4232.namprd02.prod.outlook.com/#msg58852113

**Comment by chengqi001** [comment]
> Note that there is a related email discussion (TOF for SAFIR for cylindrical scanner) at https://sourceforge.net/p/stir/mailman/stir-users/thread/BYAPR02MB42326FE5753470D1458A89ABC83F2%40BYAPR02MB4232.namprd02.prod.outlook.com/#msg58852113

Yes, he is part of our team. Hopefully, he can overcome the challenge.

**Comment by KrisThielemans** [comment]
TOF support for blocks is enabled in #1746 

The SAFIR format itself will still need changing (step 3). I have very little interest in that myself, unfortunately. (PETSIRD is a far better choice than SAFIR.)
